### PR TITLE
Travis!

### LINKED ISSUE #147: Start using Travis CI
Lets get our tests running automatically using Travis. It will probably require a bit of re-factor but we should be able to get things running on Travis easily.

**Note** While we use Docker for development we will be able to use Travis *sans-docker*, so don't feel obligated to use docker in the CI setup.

### PR DESCRIPTION
**Pull request by ElijahCaine**
fixes #147

## Changes in this PR.

- [X] Added `.travis.yml`
- [X] Setups up mysql2 database
- [X] Sets up appropriate environment variables.
- [ ] Add postgresql support.
- [X] Travis runs successfully.
- [x] Linting
- [x] Squash commits

## Testing this PR.

1. See the Travis checks?
  - Green :smile:
  - Red :sob:

@osuosl/devs

### CODE REVIEW

**Comment by ElijahCaine** [comment]
For anybody curious, I did change the version number in our Gemfile. This is b/c Travis doesn't yet support Ruby 2.3.0, but the tests run fine as long as we are running ruby 2.2.2+.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+env:
+  - MYSQL_DATABASE='test_db' \
+    MYSQL_PASSWORD='' \
+    MYSQL_ROOT_PASSWORD='' \
+    MYSQL_TESTING_HOST='localhost' \
+    MYSQL_USER='root'
+
+script:
+  - rake spec
```
Comment (alxngyn) [inline .travis.yml]: Lets add 
`rake migrate`
`rake rubocop` (this is going to be great)

we should probably add a script somewhere to do some one off things like import data and stuff

**Comment by alxngyn** [comment]
i think we have to change the `rakefile` so things fail if one of them fail. I don't think our rubocop should be passing

**Comment by ElijahCaine** [comment]
@alxngyn You're right. I'll figure out how to enforce all of our checks correctly.

File: spec/collectors/database_spec.rb [hunk 2]
```diff
@@ -8,40 +8,45 @@
...
+    # rubocop:disable VariableName
     @DB = {}
+    # rubocop:enable VariableName
 
     # Grant complete privileges to our special user
     # TODO: make this conditional, do not run if running in CI, this kind of
     # thing gets run in a pre-testing script in that situation
     # Establishes the main database connection using the given credentials.
-    @DB[:root] = Sequel.connect("mysql://root:#{ENV['MYSQL_ROOT_PASSWORD']}@#{ENV['MYSQL_TESTING_HOST']}")
+    @DB[:root] = Sequel.connect("mysql://root:#{ENV['MYSQL_ROOT_PASSWORD']}
```
Comment (ElijahCaine) [inline spec/collectors/database_spec.rb]: This needs to be
```
@DB[:root] = Sequel.connect('mysql://root:' +
                            "#{ENV['MYSQL_ROOT_PASSWORD']}" +
                            "@#{ENV['MYSQL_TESTING_HOST']}")
```

**Comment by alxngyn** [comment]
++

**Comment by Kennric** [comment]
+1

**Pull request merged**